### PR TITLE
[SE-0258] Implement revision #3 of property wrappers

### PR DESCRIPTION
**Pull request by DougGregor**
Implement the upcoming third revision of the property wrappers proposal, which includes the following changes:

* The `private` backing storage property is always named with a leading underscore, e.g., `_foo`.
* `wrapperValue` is renamed to `projectedValue`
* When `projectedValue` is present, the projection property is created (named `$foo`) at the same access level as the original property declaration.

Addresses rdar://problem/52140207

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please build toolchain

**Comment by xwu** [comment]
> `wrappedValue` is renamed to `projectedValue`

You mean `wrapperValue`

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please build toolchain macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci build toolchain

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci build toolchain

File: test/IDE/print_property_wrappers.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ func foo() -> Int { return 17 }
 struct HasWrappers {
   // CHECK: @Wrapper var x: Int {
   // CHECK-NEXT:  get
-  // CHECK: var $x: Wrapper<Int>
+  // CHECK: var _x: Wrapper<Int>
   @Wrapper(closure: foo)
   var x: Int
```
Comment (masters3d) [inline test/IDE/print_property_wrappers.swift]: What happens when there is already a property named `_x` in the same scope?

**Pull request merged**